### PR TITLE
feat: import cdc files

### LINKED ISSUE #47: Import transactions/scripts into JS rather than inlining them
In `web/src/flow`, all the transactions/scripts are inlined as strings.  I think a better option is to save them as cdc files in the `cadence/scripts` `cadence/transactions` directories and then import them into the JS.  The way I achieved this in the project I'm working on (which I forked off of `kitty-items`) is via https://github.com/pveyes/raw.macro.

So instead of:

```
const CODE = fcl.cdc`
  import FungibleToken from 0xFungibleToken
  ...
`
```

You'd have:

```
import raw from "raw.macro";;
const script = raw("../../../cadence/scripts/is_account_initialized.cdc");
const CODE = fcl.cdc`${script}`
```

The primary benefits are 1) testability (you can test your scripts/transactions in go) and 2) readability (the transactions/scripts are easier to read in cdc files than just long strings in JS files).

If the Flow team likes this idea and is willing to accept outside contributions to this repo, I'd be happy to put together a PR making this change.

### PR DESCRIPTION
**Pull request by alxflw**
This PR fixes #47 

All Cadence files that were inline are now in the one of two folders:
- `web/cadence/transactions/<name>.cdc`
- `web/cadence/scripts/<name>.cdc`

cc @shubik22

### CODE REVIEW

**Comment by psiemens** [comment]
I think you can actually get away without using `raw.macro`.

I realized we [already load Cadence files](https://github.com/onflow/fcl-dev-wallet/blob/main/pages/api/accounts/new.ts#L3) in the fcl-dev-wallet project, which is also a Next app. It works by adding this line in `next.config.js`: https://github.com/onflow/fcl-dev-wallet/blob/main/next.config.js#L23-L29

**Comment by alxflw** [comment]
> I think you can actually get away without using `raw.macro`.
> 
> I realized we [already load Cadence files](https://github.com/onflow/fcl-dev-wallet/blob/main/pages/api/accounts/new.ts#L3) in the fcl-dev-wallet project, which is also a Next app. It works by adding this line in `next.config.js`: https://github.com/onflow/fcl-dev-wallet/blob/main/next.config.js#L23-L29

that worked! but it seems the cdc files have to be within the nextjs project root. LMK if that impacts testability

**Review by 10thfloor — CHANGES_REQUESTED** [review]
Great!

**Comment by 10thfloor** [comment]
Is there is a way to de-dupe the cadence files?

**Comment by psiemens** [comment]
> Is there is a way to de-dupe the cadence files?

Unfortunately by design, Next requires that all loaded files are in the Next project root (`/web`) in this case. 

However I came across this experimental feature to load external Typescript files: https://github.com/vercel/next.js/pull/22867

Maybe it works for other file types too? Worth a try?

**Comment by alxflw** [comment]
thanks for the input y'all. I'll focus on two things:

1) experiment with loading of files below root. I'll try the webpack config approach and see if I can get something working

2) create a helper function and unify the importing style (aliases that are replaced with relative paths on load)

**Comment by alxflw** [comment]
> thanks for the input y'all. I'll focus on two things:
> 
> 1. experiment with loading of files below root. I'll try the webpack config approach and see if I can get something working
> 2. create a helper function and unify the importing style (~aliases that are replaced with relative paths on load~ relative paths replaced with addresses)

ok, I tried a few things but didn't end up with a working solution for now. details:
1) custom `next.config.js`: this is only allowing me to load files inside the nextjs root. we seem to be blocked by the nextjs issue referenced earlier. TS files can be loaded but arbitrary files cannot
2) adding a utility method that could be used across web and api projects: docker compose mounts the services (web, api, db, etc) on the respective paths, so a shared utility would require changes to that setup. I feel like this would add more complexity and won't provide a lot of benefits. we'd avoid some redundant code but it'll be hard to understand why and how it is working. devs might be more confused by that

thoughts?

**Comment by alxflw** [comment]
I'd love to get this to the finish line for initial release @psiemens @10thfloor. I'll resolve the conflicts but please confirm that we exhausted our options to simplify (prior comment)

File: web/src/flow/script.get-account-items.js [hunk 1]
```diff
@@ -1,19 +1,7 @@
...
-
-  pub fun main(address: Address): [UInt64] {
-    if let collection = getAccount(address).getCapability<&KittyItems.Collection{NonFungibleToken.CollectionPublic, KittyItems.KittyItemsCollectionPublic}>(KittyItems.CollectionPublicPath).borrow() {
-      return collection.getIDs()
-    }
-
-    return []
-  }
-`
+import FETCH_ACCOUNT_ITEMS_SCRIPT from "cadence/scripts/get_account_items.cdc"
```
Comment (psiemens) [inline web/src/flow/script.get-account-items.js:4]: Ah, I guess `fcl.cdc` isn't even needed.

I think it'd be good to remove the calls to `fcl.cdc` from all of these files to keep things simple and consistent.

File: web/cadence/scripts/get_account_items.cdc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+  import NonFungibleToken from 0xNonFungibleToken
```
Comment (psiemens) [inline web/cadence/scripts/get_account_items.cdc:1]: Indentation is a bit off in some of the Cadence files

**Review by psiemens — COMMENTED** [review]
Looks good overall! Just a few cleanup comments.

I say let's stick with this for now and then separately investigate better tooling to handle Cadence imports.

File: web/cadence/scripts/get_account_items.cdc (hunk 2 above)
Comment (alxflw) [reply web/cadence/scripts/get_account_items.cdc:1]: good point, fixing

File: web/src/flow/script.get-account-items.js (hunk 1 above)
Comment (alxflw) [reply web/src/flow/script.get-account-items.js:4]: will do

**Review by psiemens — APPROVED** [review]
LGTM! 🚢 

Congrats on first PR! 🎉

**Pull request merged**